x/cronos/keeper/evm_log_handlers.go

```diff
@@ -0,0 +1,154 @@
+package keeper
+
+import (
+	"errors"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	gravitytypes "github.com/peggyjv/gravity-bridge/module/x/gravity/types"
+
+	"github.com/crypto-org-chain/cronos/x/cronos/types"
+)
+
+var (
+	_ types.EvmLogHandler = NativeTransferHandler{}
+	_ types.EvmLogHandler = EthereumTransferHandler{}
+)
+
+var (
+	// NativeTransferEvent represent the signature of
+	// `event __CronosNativeTransfer(address recipient, uint256 amount, string denom)`
+	NativeTransferEvent abi.Event
+
+	// EthereumTransferEvent represent the signature of
+	// `event __CronosEthereumTransfer(address recipient, uint256 amount, uint256 bridge_fee)`
+	EthereumTransferEvent abi.Event
+)
+
+func init() {
+	addressType, _ := abi.NewType("address", "", nil)
+	uint256Type, _ := abi.NewType("uint256", "", nil)
+	NativeTransferEvent = abi.NewEvent(
+		"__CronosNativeTransfer",
+		"__CronosNativeTransfer",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}},
+	)
+	EthereumTransferEvent = abi.NewEvent(
+		"__CronosEthereumTransfer",
+		"__CronosEthereumTransfer",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "bridge_fee",
+			Type:    uint256Type,
+			Indexed: false,
+		}},
+	)
+}
+
+// NativeTransferHandler handles `__CronosNativeTransfer` log
+type NativeTransferHandler struct {
+	bankKeeper   types.BankKeeper
+	cronosKeeper Keeper
+}
+
+func NewNativeTransferHandler(bankKeeper types.BankKeeper, cronosKeeper Keeper) *NativeTransferHandler {
+	return &NativeTransferHandler{
+		bankKeeper:   bankKeeper,
+		cronosKeeper: cronosKeeper,
+	}
+}
+
+func (h NativeTransferHandler) EventID() common.Hash {
+	return NativeTransferEvent.ID
+}
+
+func (h NativeTransferHandler) Handle(ctx sdk.Context, contract common.Address, data []byte) error {
+	unpacked, err := NativeTransferEvent.Inputs.Unpack(data)
+	if err != nil {
+		// log and ignore
+		h.cronosKeeper.Logger(ctx).Error("log signature matches but failed to decode", "error", err)
+		return nil
+	}
+
+	denom, found := h.cronosKeeper.GetDenomByContract(ctx, contract)
+	if !found {
+		return errors.New("contract is not connected to native token")
+	}
+
+	contractAddr := sdk.AccAddress(contract.Bytes())
+	recipient := sdk.AccAddress(unpacked[0].(common.Address).Bytes())
+	coins := sdk.NewCoins(sdk.NewCoin(denom, sdk.NewIntFromBigInt(unpacked[1].(*big.Int))))
+	err = h.bankKeeper.SendCoins(ctx, contractAddr, recipient, coins)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// EthereumTransferHandler handles `__CosmosNativeGravitySend` log
```

Comment:
missing some test cases for this handler

Reply:
yeah, just added, one negative case, one positive case.